REFRAME.md

```diff
@@ -280,6 +282,40 @@ Response:
 {"PutIPNSResponse : {}"}
 ```
 
+#### Provide
+
+A message for indicating that the client is able to act as a provider for a given key.
+
+```ipldsch
+    type ProvideRequest struct 
+        Key &Any
+        Providers [Provider]
+    }

```

Comment:
From @willscott
> One design point that is implicit here but that may be worth calling out specifically, is that this format implies that providing follows the same behavior as in current IPFS implementations - that a provide has an implicit 24hr ttl.
>
> Should we have a provide / retract model like in network indexing instead?
should we support an explicit ttl?

Good questions, a few thoughts:
1. This interface cannot be used to proxy requests to the current IPFS Public DHT. This is simply a result of the current RPCs that only accept messages like "I am providing foo" rather than "Alice is providing foo"  https://github.com/libp2p/go-libp2p-kad-dht/issues/584. This means the systems most likely to use this in the shorter term are newer systems (e.g. network indexers, things that look like BitTorrent trackers, new DHTs, etc.) and so we can look toward what those systems want to use.
2. While people can use transport level protections like bearer tokens , IP censoring, etc. to protect requests like Provide from being spammed, there's nothing in the message really helping a server prevent abuse. Perhaps we want something here to help.
3. Do we want to design for these to come in bulk? I guess we could just send multiple messages after each other and rely on transport compression to reduce the overhead, so it might be fine as is.
4. Providing a recommended TTL seems pretty reasonable
5. Allowing for a retract also seems fairly reasonable, although probably there should either be no expectation from the client that this actually happens or that expectation should be communicated via the response or something (e.g. a reframe server which advertises data to two systems one it can retract from and one it cannot won't be able to fully retract it).
     - Note; Retract seems even more problematic than the rest without some authentication/abuse protection that prevents me from retracting your content just because I feel like it (e.g. I ask for providers then retract them all)